core/sql/executor/ExExeUtilLoad.cpp

```diff
@@ -2890,7 +2890,12 @@ short ExExeUtilLobExtractTcb::work()
 
 	      }
 	    
-	    step_ = RETRIEVE_LOB_LENGTH_;	    
+	    if (lobTdb().getToType() == ComTdbExeUtilLobExtract::TO_BUFFER_)
+	      step_ = EXTRACT_LOB_DATA_;
+	    else
+	      if ((lobTdb().getToType() == ComTdbExeUtilLobExtract::RETRIEVE_LENGTH_) || (lobTdb().getToType() == ComTdbExeUtilLobExtract::TO_FILE_))
+	      step_ = RETRIEVE_LOB_LENGTH_;
+	    
```

Comment:
So, if neither of these 'ifs' are true, do we just remain in the same step_? Could that be an infinite loop?